dask/array/core.py

```diff
@@ -949,6 +949,17 @@ class Array(Base):
     _default_get = staticmethod(threaded.get)
     _finalize = staticmethod(finalize)
 
+    def __new__(cls, *args, **kwargs):
+        obj = super(Array, cls).__new__(cls)
+        cls.__init__(obj, *args, **kwargs)
```

Comment:
Perhaps we could make `__init__` short circuit if the object already has a `.dask` attribute.